src/channel.c

```diff
@@ -80,52 +78,94 @@ IMPLEMENT_MOCKABLE_FUNCTION(, THANDLE(CHANNEL), channel_create, THANDLE(PTR(LOG_
     return result;
 }
 
-IMPLEMENT_MOCKABLE_FUNCTION(, CHANNEL_RESULT, channel_pull, THANDLE(CHANNEL), channel, THANDLE(RC_STRING), correlation_id,PULL_CALLBACK, pull_callback, void*, pull_context, THANDLE(ASYNC_OP)*, out_op_pull)
+IMPLEMENT_MOCKABLE_FUNCTION(, int, channel_open, THANDLE(CHANNEL), channel)
+{
+    int result;
+    /*Codes_SRS_CHANNEL_43_095: [If channel is NULL, channel_open shall fail and return a non - zero value.]*/
+    if (channel == NULL)
+    {
+        LogError("Invalid arguments: THANDLE(CHANNEL) channel=%p", channel);
+        result = MU_FAILURE;
+    }
+    else
+    {
+        CHANNEL* channel_ptr = THANDLE_GET_T(CHANNEL)(channel);
+        /*Codes_SRS_CHANNEL_43_096 : [channel_open shall call channel_internal_open.]*/
+        if (channel_internal_open(channel_ptr->channel_internal) != 0)
+        {
+            /*Codes_SRS_CHANNEL_43_099: [ If there are any failures, channel_open shall fail and return a non-zero value. ]*/
+            LogError("Failure in channel_internal_open(channel_internal=%p)", channel_ptr->channel_internal);
+            result = MU_FAILURE;
+        }
+        else
+        {
+            result = 0;
+        }
+    }
+    return result;
+}
+
+IMPLEMENT_MOCKABLE_FUNCTION(, void, channel_close, THANDLE(CHANNEL), channel)
+{
+    /*Codes_SRS_CHANNEL_43_097: [If channel is NULL, channel_close shall return immediately.]*/
+    if (channel == NULL)
+    {
+        LogError("Invalid arguments: THANDLE(CHANNEL) channel=%p", channel);
+    }
+    else
+    {
+        CHANNEL* channel_ptr = THANDLE_GET_T(CHANNEL)(channel);
```

Comment:
>CHANNEL* channel_ptr = THANDLE_GET_T(CHANNEL)(channel); [](http://example.com/codeflow?start=8&length=55)

you shouldn't need the GET_T here, right?

Reply:
Fixed.